packages/docregistry/src/context.ts

```diff
@@ -95,6 +95,13 @@ export class Context<T extends DocumentRegistry.IModel>
     return this._fileChanged;
   }
 
+  /**
+   * A signal emitted on the start and end of a saving operation
+   */
+  get saving(): ISignal<this, DocumentRegistry.SaveState> {
```

Comment:
While we are on semantics, I think i would call this `saveState` and the private var `_saveState`.